pkg/controller/machine/controller_test.go

```diff
@@ -37,6 +37,9 @@ func TestReconcileRequest(t *testing.T) {
 			Name:       "create",
 			Namespace:  "default",
 			Finalizers: []string{v1alpha1.MachineFinalizer},
+			Labels: map[string]string{
```

Comment:
I'm wondering if we should use an annotation here rather than a label, that said, it would require modifying the clusterctl logic related to determining if the controlplane machine is "ready".

Reply:
The issue mentions "stronger" link, which I guess a weak one is stronger than no link. Although I'm wondering if we should make this a field in the MachineSpec instead of a label. What are pros for annotations instead?

Reply:
Getting consensus on a field in the MachineSpec might be difficult, but annotations are not as user visible and less likely for users to modify.

Reply:
Got it. I'm thinking though that labels might be more useful to query for example machines in a cluster, where annotations is mostly extra metadata.

Reply:
@vincepri according to the k8s docs, Lables are not meant to be meaningful and relevant to users, but do not directly imply semantics: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/

While one of the uses for annotations is said to be "Directives from the end-user to the implementations to modify behavior or engage non-standard features": https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/

Since the presence (or absence) of the label changes behavior in the underlying implementation, it would seem to fit more with the annotation use case rather than the label use case.

Reply:
`kubeadm` uses a label to mark master nodes. Since labels are searchable, maybe they are better for [identifying information](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#attaching-metadata-to-objects):

> You can use either labels or annotations to attach metadata to Kubernetes objects. Labels can be used to select objects and to find collections of objects that satisfy certain conditions. In contrast, annotations are not used to identify and select objects.